### PR TITLE
2015 07 06 triton r1

### PR DESCRIPTION
**Pull request by misterbisson**
This PR implements the [clustered Couchbase in Docker containers demo](https://www.joyent.com/blog/couchbase-in-docker-containers) ([in git](https://github.com/misterbisson/clustered-couchbase-in-containers)) in Mesos.
- Consul is promoted up to the level of Mesos as a shared resource. This makes discovery problems in Mesos easier to resolve.
  - TODO: the linking of Docker containers to Consul is janky. I'd love to see this work as cleanly as it does in Docker Compose + Triton, but it does't (today). The next best solution would be to use DNS, but that requires injecting DNS settings into the container. I'd love for Mesos + Marathon to offer better support for this. Investigation into options is needed for both cases.
- The Couchbase container and a simple client app are defined as separate Marathon tasks. I'd originally defined them as an application group, but then realized that they need separate lifecycles. We don't want Mesos + Marathon to redeploy Couchbase (and lose data from the filesystem in the running containers) when we release a new version of the client, for example
  - Unfortunately, it's not safe to **scale down** Couchbase purely in Mesos + Marathon as this demo works now. The issue is that Couchbase needs to rebalance data into a smaller set of nodes before destroying excess nodes
- The repo includes a `start.sh` script that adds prompts for everything. Get started using the instructions in the existing [README.md](https://github.com/joyent/mesos-dockerfiles/blob/master/README.md)

### CODE REVIEW

**Comment by dekobon** [comment]
LGTM

**Comment by xer0x** [comment]
The couchbase demo on Mesos script gets stuck for me. #4

File: marathon-tasks/couchbase.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{
+    "id": "/couchbase",
+    "cpus": 2,
+    "mem": 4096,
+    "container": {
+        "type": "DOCKER",
+        "volumes": [],
+        "docker": {
+            "image": "misterbisson/triton-couchbase:enterprise-3.0.3-r2",
```
Comment (xer0x) [inline marathon-tasks/couchbase.json:9]: Should this be 3.0.3-r3?  #4

Comment (misterbisson) [inline marathon-tasks/couchbase.json:9]: Fixed in https://github.com/joyent/mesos-dockerfiles/pull/5

**Pull request merged**